src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1751,7 +1752,47 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             _terminal->MultiClickSelection(terminalPosition, mode);
             selectionNeedsToBeCopied = true;
         }
-        _updateSelectionUI();
+        else if (_settings->MoveCursorWithMouse()) // This is also mode==Char && !shiftEnabled
+        {
+            // If we're handling a single left click, without shift pressed, and
+            // outside mouse mode, AND the user has MoveCursorWithMouse turned
+            // on, let's try to move the cursor.
+            //
+            // We'll only move the cursor if the user has clicked after the last
+            // mark, if there is one. That means the user also needs to set up
+            // shell integration to enable this feature.
+            //
+            // As noted in GH #8573, there's plenty of edge cases with this
+            // approach, but it's good enough to bring value to 90% of use cases.
+            const auto cursorPos{ _terminal->GetCursorPosition() };
+
+            // Does the current buffer line have a mark on it?
+            const auto& marks{ _terminal->GetScrollMarks() };
+            if (!marks.empty())
+            {
+                const auto& last{ marks.back() };
+                const auto [start, end] = last.GetExtent();
+                if (terminalPosition >= end)
+                {
+                    // Get the distance between the cursor and the click, in cells.
+                    const auto bufferSize = _terminal->GetTextBuffer().GetSize();
+                    const auto delta = bufferSize.CompareInBounds(terminalPosition, cursorPos);
+                    const WORD key = delta > 0 ? VK_RIGHT : VK_LEFT;
+                    const auto keystrokes = std::abs(delta);
+                    // Send an up and a down once per cell. This won't
+                    // accurately handle wide characters, or continuation
+                    // prompts, or cases where a single escape character in the
+                    // command (e.g. ^[) takes up two cells.
+                    for (auto i = 0; i < keystrokes; i++)
+                    {
+                        _terminal->SendKeyEvent(key, 0, {}, true);
+                        _terminal->SendKeyEvent(key, 0, {}, false);

```

Comment:
For purposes like this it would be nice if we could stop flushing our input pipe so that the keystrokes are sent as a unison. For both pwsh and cmd this would avoid re-printing the entire commandline on every cursor navigation. One option would be to do something like `_terminal->CorkInputPipe()` and `UncorkInputPipe()` (borrowing `TCP_CORK` terminology), but I also somewhat feel like it would be better if we instead made it so that `SendKeyEvent` will never flush the input pipe and instead we add a `_terminal->FlushInput()` function which you need to call when you're done generating input. Alternatively, we could also extract `TerminalInput` out of `Terminal`s control and accumulate a `std::string` here before sending it off. I don't think there's any fantastic solution, but there's a lot of alright ones. 🙂

Reply:
Could we create a follow-up issue for this? I think it would be worthwhile in general to keep track of features that would benefit of corking/uncorking the input pipe.